python/ray/tune/ray_trial_executor.py

```diff
@@ -126,6 +127,21 @@ def logger_creator(config):
     def _train(self, trial):
         """Start one iteration of training and save remote id."""
 
+        if self._find_item(self._paused, trial):
+            raise TuneError(
+                "Should not call `train` on PAUSED trial {}. "
+                "This is an internal error - please file an issue "
+                "on https://github.com/ray-project/ray/issues/.".format(
+                    str(trial)))
```

Comment:
Is there any logging information from trial that we might want to have in case this occurs? If so, might be useful to include that here.

Reply:
Hm, not sure if there's anything inherent to the trial object itself here.

I think that logging should happen in another class (like the scheduler or runner) because that's where the decision-making is done.